manifests/node/kubelet.pp

```diff
@@ -135,21 +136,13 @@
     'cgroupDriver'       => 'systemd',
   } + $_authentication_hash
 
-  file { '/etc/modules-load.d/k8s':
-    ensure  => $ensure,
-    content => file('k8s/etc/modules-load.d/k8s'),
-  }
-  exec {
-    default:
-      path        => ['/bin', '/sbin', '/usr/bin'],
-      refreshonly => true,
-      subscribe   => File['/etc/modules-load.d/k8s'];
-
-    'modprobe overlay':
-      unless => 'lsmod | grep overlay';
-
-    'modprobe br_netfilter':
-      unless => 'lsmod | grep overlay';
+  if $manage_kernel_modules {
+    kmod::load { 'overlay':
+      ensure  => $ensure,
+    }
+    kmod::load { 'br_netfilter':
+      ensure  => $ensure,
+    }
```

Comment:
Just to keep things consistent, I'l prefer to see this in the form of;

```puppet
kmod::load {
  default:
    ensure => $ensure;

  'overlay':;
  'br_netfilter':;
}
```

Reply:
Can also be done with iteration like this:

```suggestion
    [
      'br_netfilter',
      'overlay',
    ].each |String[1] $mod| {
      kmod::load { $mod:
        ensure => $ensure,
      }
    }
```

Reply:
Certainly possible, but I was more commenting on keeping things consistent compared to how it's done everywhere else in the module.

Reply:
why so complicated? We can write it even shorter:
```puppet
kmod::load { ['br_netfilter', 'overlay']:
  ensure => $ensure,
}
```

Reply:
I like @bastelfreak s solution. But I will go with @ananace proposal. So it stays consistent. I can make a separate PR to change all resources later.

Reply:
@bastelfreak that style is "bad" according to the style guide ;) https://www.puppet.com/docs/puppet/7/style_guide.html#multiple-resources